tree/treeplayer/inc/ROOT/TDFActionHelpers.hxx

```diff
@@ -511,8 +517,10 @@ public:
       if (!fDirName.empty()) {
          treeDirectory = fOutputFiles[slot]->mkdir(fDirName.c_str());
       }
-      fOutputTrees[slot] = new TTree(fTreeName.c_str(), fTreeName.c_str(), /*splitlvl=*/99, /*dir=*/treeDirectory);
+      fOutputTrees[slot] = new TTree(fTreeName.c_str(), fTreeName.c_str(), fOptions.splitlevel, /*dir=*/treeDirectory);
       fOutputTrees[slot]->ResetBit(kMustCleanup); // do not mingle with the thread-unsafe gListOfCleanups
+      if (fOptions.autosave)
+         fOutputTrees[slot]->SetAutoSave(fOptions.autosave);
```

Comment:
same as before, why cutting out the negative values?

Reply:
This only avoids setting autosave when it's zero, you can set it to a negative value.

Reply:
I don't know how ints convert to booleans :sweat: